### ISSUE TITLE
aria-allowed-attr possible regression?

### ISSUE THREAD

**Issue by thepassle** [dequelabs/axe-core#3241]
Product: axe-core

Expectation: element with both `aria-controls` and `aria-expanded` passes

Actual: element with both `aria-controls` and `aria-expanded` fails

Motivation: I believe it might have been an accidental regression via https://github.com/dequelabs/axe-core/blob/develop/CHANGELOG.md#:~:text=aria-allowed-attr%3A%20check%20for%20invalid

<pre><code>
axe-core version: 4.3.4
</code></pre>

We recently found a test started failing that uses axe-core. We previously had `axe-core@4.3.3` installed, where the test passes, the test has started failing since `axe-core@4.3.4`, which leads me to believe it may have been an accidental regression via this change: https://github.com/dequelabs/axe-core/blob/develop/CHANGELOG.md#:~:text=aria-allowed-attr%3A%20check%20for%20invalid

For context, we have an element with both an `aria-controls` and `aria-expanded` attribute, e.g.:
```html
<my-el aria-controls="menu" aria-expanded="false"></my-el>
```
As far as my accessibility knowledge goes, this should be valid (but I'm happy to be corrected if I'm wrong). I also checked with a couple of colleagues, who also expected this to be valid.

**Comment by WilcoFiers** [comment]
We've released 4.3.5, which has disabled this test. It wasn't meant to go out in a patch, we had it planned for the 4.4.0 release. We're going to take a closer look at how this can affect custom elements. There may be ways in which this can be alright, but Steve's right that these attributes don't work just by themselves. Either you reflect them onto another element, or you make that particular component fully accessible (providing role and focus).

**Comment by svanherk** [comment]
If it helps, we were also hit by this problem and use these attributes on custom elements.  We do what you described above - our custom button that opens a dropdown is written like this:
```
<custom-button aria-expanded="true"></custom-button>
```
And turns into this:
```
<custom-button aria-expanded="true">
  #shadow-root
    <button aria-expanded="true"></button>
</custom-button>
```
As you said, we really shouldn't have been doing this but it was interpreted correctly by the screenreader. For our newer components we've been using `data-aria-expanded` or just `expanded` on the custom element instead.  We have a number of older components that will need to be updated and fixed throughout our application, so we'll be blocked from going to `4.4.0` until we do that.

**Comment by straker** [comment]
@svanherk Thanks for letting us know. We'll try to come up with a better solution for 4.4.0 taking that into account.

**Comment by WilcoFiers** [comment]
I'm leaning towards flagging unknown elements in this rule for review, except if they have a valid role or a tabindex attribute. Those kinds of elements we just don't know enough about what's going on. This seems like the safest way to test them.

**Comment by straker** [comment]
A followup on this issue. Axe-core 4.4 (releasing at the end of this month) will add back the violation for non-global ARIA attributes on elements with a role. We have also introduced a fix for custom elements to report as Needs Review rather than a failure.
